OpenWeatherMap.Standard/Implementations/RestServiceCaller.cs

```diff
@@ -1,55 +1,94 @@
-﻿using OpenWeatherMap.Standard.Models;
-using System;
-using System.Collections.Generic;
+﻿using System;
+using System.Diagnostics;
 using System.Net.Http;
-using System.Text;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
+using OpenWeatherMap.Standard.Interfaces;
+using OpenWeatherMap.Standard.Models;
 
 namespace OpenWeatherMap.Standard.Implementations
 {
     /// <summary>
-    /// "default" implementation of IRestService
+    ///     "default" implementation of IRestService
     /// </summary>
-    internal class RestServiceCaller : Interfaces.IRestService
+    internal class RestServiceCaller : IRestService
     {
         private static HttpClient httpClient;
+
         /// <summary>
-        /// the HttpClient to be used
+        ///     the HttpClient to be used
         /// </summary>
         private static HttpClient HttpClient
         {
             get
             {
                 if (httpClient == null)
-                    httpClient = new HttpClient() { Timeout = TimeSpan.FromSeconds(10) };
+                    httpClient = new HttpClient {Timeout = TimeSpan.FromSeconds(10)};
                 return httpClient;
             }
         }
 
         /// <summary>
-        /// get data and deserialize them
+        ///     get data and deserialize them
         /// </summary>
         /// <param name="url">the url to perform the request</param>
+        /// <param name="iconDataBaseUrl">the root url to the weather icons</param>
         /// <returns>WeatherData object</returns>
-        public async Task<WeatherData> GetAsync(string url)
+        public async Task<WeatherData> GetAsync(string url, string iconDataBaseUrl)
```

Comment:
Could we add a default value to iconDataBaseUrl of http://openweathermap.org/img/wn/ so it does not break existing users of the library?

Reply:
Sure, I was a bit too quick with this one as I forgot one might use it's own implementation.
I'm not really happy with my new methods anyway cause I don't know where to place it best. I don't want to break your whole design so maybe you have a better idea where to put them.

Reply:
Just checked my changes in, maybe you want take a look at my repo. It includes the forecast-data and makes the icon-fetching optional.
Is there any special reason why you return a Task on your synchronous methods?
My synchronous forecast-functions execute in a separate thread-pool thread to avoid any deadlocks so I think this is a better approach cause the caller doesn't expect a Task from a sync. method?!

Reply:
The method returns a task so it can use await.  This is the typical way to make a call that can take a little while without bocking the ui of a windows or phone app

Reply:
The code looks ok

Reply:
> This is the typical way to make a call that can take a little while without bocking the ui of a windows or phone app

Yeah, but if you don't want to block the UI use the async methods with await. A synchronous-method is intended to block. But hey, it's your codebase, just change it as you like :)
There's are big discussion on this topic here [https://stackoverflow.com/questions/9343594/how-to-call-asynchronous-method-from-synchronous-method-in-c](url) and I'm no expert and just use some recommended approach which might not be the best in every scenario!

As you can see, ReSharper "fixes" lots of cosmetics, some mismatching namespaces and other stuff in your code, sorry about that. Maybe you want to merge the changes to a new, different codebase if there are too much changes. But I think none of them breaks or changes existing usage/functionality.

Will submit a PR later, I'm currently afk.

Reply:
Ok thanks will read the thread later tonight